pkg/kn/commands/service/wait.go

```diff
@@ -0,0 +1,89 @@
+// Copyright © 2019 The Knative Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package service
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	"knative.dev/client/pkg/kn/commands"
+	clientservingv1 "knative.dev/client/pkg/serving/v1"
+)
+
+var waitExample = `
+  # Waits on a service 'svc'
+  kn service wait svc
+
+  # Waits on a service 'svc' with a timeout
+  kn service wait svc --wait-timeout 10
+
+  # Waits on a service 'svc' with a timeout and wait window
+  kn service wait svc --wait-timeout 10 --wait-window 1`
+
+// NewServiceWaitCommand represents 'kn service wait' command
+func NewServiceWaitCommand(p *commands.KnParams) *cobra.Command {
+	var waitFlags commands.WaitFlags
+	command := &cobra.Command{
+		Use:     "wait NAME",
+		Short:   "Wait for a service to be ready",
+		Example: waitExample,
+		RunE: func(cmd *cobra.Command, args []string) (err error) {
+			if len(args) != 1 {
+				return errors.New("'service wait' requires the service name given as single argument")
+			}
+
+			namespace, err := p.GetNamespace(cmd)
+			if err != nil {
+				return err
+			}
+			client, err := newServingClient(p, namespace, "")
+			if err != nil {
+				return err
+			}
+
+			name := args[0]
+			out := cmd.OutOrStdout()
+
+			fmt.Fprintf(out, "Waiting for Service '%s' in namespace '%s':\n", args[0], namespace)
+			fmt.Fprintln(out, "")
+			wconfig := clientservingv1.WaitConfig{
+				Timeout:     time.Duration(waitFlags.TimeoutInSeconds) * time.Second,
+				ErrorWindow: time.Duration(waitFlags.ErrorWindowInSeconds) * time.Second,
+			}
+			err = waitForService(cmd.Context(), client, name, out, wconfig)
+			if err != nil {
+				return err
+			}
+			fmt.Fprintln(out, "")
+			fmt.Fprintf(out, "Service '%s' in namespace '%s' is ready.\n", name, namespace)
+
+			return nil
+
+		},
+	}
+	commands.AddNamespaceFlags(command.Flags(), false)
+	waitTimeoutDefault := commands.WaitDefaultTimeout
+
+	timeoutUsage := "Seconds to wait before giving up on waiting for service to be ready."
+	command.Flags().IntVar(&waitFlags.TimeoutInSeconds, "wait-timeout", waitTimeoutDefault, timeoutUsage)
+
+	windowUsage := "Seconds to wait for service to be ready after a false ready condition is returned"
+	command.Flags().IntVar(&waitFlags.ErrorWindowInSeconds, "wait-window", 2, windowUsage)
```

Comment:
Could you pls look at extracting those flags in a separate function in `wait_flags.go` to be usable without `--wati`? To keep the names and descriptions if any change happens etc.

https://github.com/knative/client/blob/main/pkg/kn/commands/wait_flags.go#L52-L56

Reply:
I changed it by reusing the current function in `wait_flags.go`